src/main/kotlin/no/nav/syfo/infrastructure/kafka/identhendelse/IdenthendelseConsumer.kt

```diff
@@ -0,0 +1,35 @@
+package no.nav.syfo.infrastructure.kafka.identhendelse
+
+import kotlinx.coroutines.delay
+import no.nav.syfo.infrastructure.kafka.KafkaConsumerService
+import org.apache.avro.generic.GenericRecord
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+class IdenthendelseConsumer(private val identhendelseService: IdenthendelseService) : KafkaConsumerService<GenericRecord> {
+    override val pollDurationInMillis: Long = 1000
+
+    override suspend fun pollAndProcessRecords(kafkaConsumer: KafkaConsumer<String, GenericRecord>) {
+        runCatching {
+            val records = kafkaConsumer.poll(Duration.ofMillis(pollDurationInMillis))
+            if (records.count() > 0) {
+                records.mapNotNull { it.value() }.forEach {
+                    identhendelseService.handle(identhendelse = it.toKafkaIdenthendelseDTO())
+                }
+                kafkaConsumer.commitSync()
+            }
+        }.onFailure { ex ->
```

Comment:
Kan kanskje forenkle likegodt? (dvs ta bort runCatching og onFailure - hvis den stopper vil jo pod'en restarte.

Reply:
Hmja, nå som vi ikke gjør kall til PDL for å sjekke om identen er oppdatert er det kanskje ingen grunn til å unsubscribe, vente og prøve på nytt...

Reply:
Ja, akkurat (tror ikke denne consumeren trenger å oppføre seg noen annerledes enn alle andre vi har).